windows.js

```diff
@@ -65,230 +37,191 @@ export async function install(platform, engine, version) {
 
-    let url = `${msys2GCCReleaseURI}/msys2-gcc-pkgs/msys2${suffix}.7z`
-    console.log(url)
-    return await tc.downloadTool(url)
-  })
+  const parentDir = path.dirname(rubyPrefix)
+  if (!fs.existsSync(parentDir)) {
+    fs.mkdirSync(parentDir, { recursive: true })
+  }
 
-  // need to remove all directories, since they may indicate old packages are installed,
-  // otherwise, error of "error: duplicated database entry"
-  fs.rmSync(`${msys2BasePath}\\var\\lib\\pacman\\local`, { recursive: true, force: true })
+  const extractedPath = path.join(extractPath, base)
+  if (extractedPath[0] === rubyPrefix[0]) {
+    fs.renameSync(extractedPath, rubyPrefix)
+  } else {
+    fs.symlinkSync(extractedPath, rubyPrefix, 'junction')
+  }
 
-  await common.measure(`Extracting  msys2 build tools`, async () =>
-    exec.exec('7z', ['x', downloadPath, '-aoa', '-bd', `-o${msys2BasePath}`], { silent: true }))
+  if (common.shouldUseToolCache(engine, version)) {
+    common.createToolCacheCompleteFile(rubyPrefix)
+  }
 }
 
-// Windows JRuby can install gems that require compile tools, only needed for
-// windows-2022 and later images
 export async function installJRubyTools() {
-  await installMSYS2Tools()
-  await installGCCTools(os.arch() === 'arm64' ? 'clangarm64' : 'mingw64', '3.4')
-}
-
-// Install vcpkg files needed to build mswin Ruby
-async function installVCPkg() {
-  const downloadPath = await common.measure(`Downloading mswin vcpkg packages`, async () => {
-    let url = `${msys2GCCReleaseURI}/msys2-gcc-pkgs/mswin.7z`
-    console.log(url)
-    return await tc.downloadTool(url)
-  })
-
-  await common.measure(`Extracting  mswin vcpkg packages`, async () =>
-    exec.exec('7z', ['x', downloadPath, '-aoa', '-bd', `-o${vcPkgBasePath}`], { silent: true }))
+  return await installMSYS2(toolchainVersions['head'][os.arch()])
 }
 
-async function downloadAndExtract(engine, version, url, base, rubyPrefix) {
-  const parentDir = path.dirname(rubyPrefix)
-
-  const downloadPath = await common.measure('Downloading Ruby', async () => {
+async function installMSYS2(url, rubyPrefix = process.env.RUNNER_TEMP) {
+  const downloadPath = await common.measure('Downloading msys2 build tools', async () => {
     console.log(url)
     return await tc.downloadTool(url)
   })
 
-  await common.measure('Extracting  Ruby', async () =>
-    // -xr extract but exclude share\doc files
-    exec.exec('7z', ['x', downloadPath, '-bd', `-xr!${base}\\share\\doc`, `-o${parentDir}`], { silent: true }))
+  const extractPath = path.join(process.env.RUNNER_TEMP, 'msys64')
+  await common.measure('Extracting  msys2 build tools', async () =>
+    exec.exec('7z', ['x', downloadPath, '-aoa', '-bd', `-o${extractPath}`], { silent: true }))
 
-  if (base !== path.basename(rubyPrefix)) {
-    await io.mv(path.join(parentDir, base), rubyPrefix)
+  const msys2Path = path.join(rubyPrefix, 'msys64')
```

Comment:
Could you add a code comment here about the choice of rubyPrefix/msys64, like you explained in the PR description?